src/SpotifyRemote.ts

```diff
@@ -253,20 +252,97 @@ export interface SpotifyRemoteApi extends TypedEventEmitter<SpotifyRemoteEvents>
     getCrossfadeState(): Promise<CrossfadeState>;
 }
 
-/**
- * @ignore
- */
-const SpotifyRemote = NativeModules.RNSpotifyRemoteAppRemote as SpotifyRemoteApi;
-RNEvents.register(SpotifyRemote);
-RNEvents.conform(SpotifyRemote);
-
-// Example of Javascript only api method
-SpotifyRemote.setPlaying = (playing: boolean) => {
-    // todo: Will want to likely check the state of playing somewhere?
-    // Perhaps this can be done in native land so that we don't need to
-    // worry about it here
-    return playing ? SpotifyRemote.resume() : SpotifyRemote.pause();
-}
+const nativeModule = NativeModules.RNSpotifyRemoteAppRemote;
+
+const nativeEventEmitter = new NativeEventEmitter(nativeModule);
+const eventListeners: Record<
+    keyof SpotifyRemoteEvents,
+    Set<EmitterSubscription>
+> = {
+    playerContextChanged: new Set(),
+    playerStateChanged: new Set(),
+    remoteConnected: new Set(),
+    remoteDisconnected: new Set(),
+};
+
+const SpotifyRemote: SpotifyRemoteApi = {
+    // Native APIs
+    connect: nativeModule.connect.bind(nativeModule),
+    disconnect: nativeModule.disconnect.bind(nativeModule),
+    getChildrenOfItem: nativeModule.getChildrenOfItem.bind(nativeModule),
+    getContentItemForUri: nativeModule.getContentItemForUri.bind(nativeModule),
+    getCrossfadeState: nativeModule.getCrossfadeState.bind(nativeModule),
+    getPlayerState: nativeModule.getPlayerState.bind(nativeModule),
+    getRecommendedContentItems:
+        nativeModule.getRecommendedContentItems.bind(nativeModule),
+    getRootContentItems: nativeModule.getRootContentItems.bind(nativeModule),
+    isConnectedAsync: nativeModule.isConnectedAsync.bind(nativeModule),
+    pause: nativeModule.pause.bind(nativeModule),
+    playItem: nativeModule.playItem.bind(nativeModule),
+    playItemWithIndex: nativeModule.playItemWithIndex.bind(nativeModule),
+    playUri: nativeModule.playUri.bind(nativeModule),
+    queueUri: nativeModule.queueUri.bind(nativeModule),
+    resume: nativeModule.resume.bind(nativeModule),
+    seek: nativeModule.seek.bind(nativeModule),
+    setRepeatMode: nativeModule.setRepeatMode.bind(nativeModule),
+    setShuffling: nativeModule.setShuffling.bind(nativeModule),
+    skipToNext: nativeModule.skipToNext.bind(nativeModule),
+    skipToPrevious: nativeModule.skipToPrevious.bind(nativeModule),
+    setPlaying(playing: boolean) {
+        // todo: Will want to likely check the state of playing somewhere?
+        // Perhaps this can be done in native land so that we don't need to
+        // worry about it here
+        return playing ? this.resume() : this.pause();
+    },
+    // Listeners
+    addListener(eventType, listener) {
+        const sub = nativeEventEmitter.addListener(eventType, listener);
+        if (this.listenerCount(eventType) === 0) {
+            nativeModule.eventStartObserving(eventType);
+        }
+        eventListeners[eventType].add(sub);
+        return this;
```

Comment:
If we return the `sub` here.  Then this method looks exactly like the nativeEventEmitter and we can add deprecations to `removeListener` below similar to here 
https://github.com/facebook/react-native/blob/2596b2f6954362d2cd34a1be870810ab90cbb916/Libraries/vendor/emitter/_EventEmitter.js#L82

Reply:
Done via https://github.com/cjam/react-native-spotify-remote/commit/b6abe5bfe8418da7fd13c6edbdd91ebe766ff18c